singer_sdk/streams/core.py

```diff
@@ -963,13 +967,27 @@ def reset_state_progress_markers(self, state: dict | None = None) -> None:
 
         reset_state_progress_markers(state)
 
+    def _finalize_state_progress_markers(
```

Comment:
Can we rename this? There's now three functions with the same name and signature, so it's hard to tell what each of them is doing:

* `singer_sdk.helpers._state.finalize_state_progress_markers`
* `singer_sdk.streams.core.Stream.finalize_state_progress_markers`
* `singer_sdk.streams.core.Stream._finalize_state_progress_markers`

Reply:
Done in 9c353cce4b1b2d886d4463b9ef24630f932c76fe